lib/rubygems/commands/push_command.rb

```diff
@@ -48,22 +48,25 @@ def initialize
   end
 
   def execute
-    gem_name = get_one_gem_name
-    default_gem_server, push_host = get_hosts_for(gem_name)
-
-    @host = if @user_defined_host
-              options[:host]
-            elsif default_gem_server
-              default_gem_server
-            elsif push_host
-              push_host
-            else
-              options[:host]
-            end
-
-    sign_in @host, scope: get_push_scope
-
-    send_gem(gem_name)
+    gem_names = get_all_gem_names
+
+    default_gem_server, push_host = get_hosts_for(gem_names.first)
+
+      @host = if @user_defined_host
+                options[:host]
+              elsif default_gem_server
+                default_gem_server
+              elsif push_host
+                push_host
+              else
+                options[:host]
+              end
+
+      sign_in @host, scope: get_push_scope
+
+    gem_names.each do |gem_name|
+      send_gem(gem_name)
```

Comment:
ToDo: Rescue and retry any failures. Maybe even add an option to skip a failing gem and go on to the next one?